lib/fastlane/actions/sync_build_number_to_git.rb

```diff
@@ -0,0 +1,38 @@
+module Fastlane
+  module Actions
+    class SyncBuildNumberToGitAction < Action
+      def self.is_git?
+        Actions.sh 'git rev-parse HEAD'
+        return true
+      rescue
+        return false
+      end
+
+      def self.run(params)
+        if is_git?
+          command = 'git rev-list HEAD --count'
+        else
+          raise "Not in a git repository."
+        end
+        build_number = (Actions.sh command).strip
+        Fastlane::Actions::IncrementBuildNumberAction.run(build_number: build_number)
```

Comment:
Please add some kind of error handling here, in case `build_number` is not valid.